### DISCUSSION TITLE
ios微信端重连问题

### DISCUSSION THREAD

**Discussion by Combox722** [bytedance/xgplayer#835] [Q&A] [answered]
**您使用的西瓜播放器版本是多少？ What version of xgplayer are you using?**
xgplayer: 2.32.1
xgplayer-hls.js: 2.6.3

**您使用的操作系统和浏览器分别是？ What OS and browser are you using?**
ios 15.4 微信端 

**如何复现问题？ What did you do?**
obs推流中断3-5s后恢复

**您期望的播放器正常行为是？ What did you expect to happen?**
推流/网络中断后播放器出现loading加载，恢复推流后继续播放

**实际播放器的表现是？ What actually happened?**
推流中断后触发waiting事件后显示loading圈，接着直接触发了playing事件loading圈消失（此时直播流还未恢复），停留在最后一帧画面（长时间后会黑屏），正常推流后需要很长时间才能恢复或者一直恢复不了

应用了xgplayer-hls.js 但无法生效（ios 判断Hls.isSupported是不支持的）还是原生video的能力播放m3u8，这种重连情况是否可以处理？或者异常的事件抛出导致的无法准确判断播放器状态有更好的办法解决吗

**Comment by gemxx** [comment]
1. iOS下没有MediaSource，因此无法使用hls.js，会直接用video。
2. 使用原生video的情况下，可以监听video waiting或者error事件进行重试。重试可采用重新设置video source重新开播

**Reply by trry-hub** [reply]
现在有好的解决方案了么？

**Reply by gemxx** [reply, marked as answer]
> 现在有好的解决方案了么？

现在iOS 17+支持MMS (受控MSE， buffer缓冲可以被浏览器管控，默认低水位10s、高水位30s), xgplayer-hls、xgplayer-flv都支持了。需要在插件中增加配置 `preferMMS: true` 以启用
